components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -162,33 +171,82 @@ async def do_search(
     begin_timestamp: int | None,
     end_timestamp: int | None,
     ignore_case: bool,
-    max_num_results: int,
     path_filter: str | None,
     do_count_aggregation: bool | None,
     count_by_time_bucket_size: int | None,
 ):
-    db_monitor_task = asyncio.ensure_future(
-        run_function_in_process(
-            create_and_monitor_job_in_db,
-            db_config,
-            results_cache,
-            wildcard_query,
-            tags,
-            begin_timestamp,
-            end_timestamp,
-            ignore_case,
-            max_num_results,
-            path_filter,
-            do_count_aggregation,
-            count_by_time_bucket_size,
+    if do_count_aggregation is None:
+        host = None
+        for ip in set(socket.gethostbyname_ex(socket.gethostname())[2]):
```

Comment:
Why not `gethostbyname` instead? (Since we're not doing anything with the extra info that `gethostbyname_ex` returns.)

Reply:
Because `gethostbyname` just returns `127.0.0.1`, but the search worker and the client may not run in the same node?

Reply:
`gethostbyname_ex` can't return `127.0.0.1`? Or did you mean to loop over the IPs returned by `gethostbyname_ex` to find one which isn't `127.0.0.1`?

Reply:
`socket.gethostbyname_ex(socket.gethostname())[0]` is actually `127.0.0.1` and `[2]` is the other IP of the host.

Reply:
Uh, correct me if I'm wrong, but `gethostbyname_ex` [returns](https://docs.python.org/3/library/socket.html#socket.gethostbyname_ex) "a 3-tuple (hostname, aliaslist, ipaddrlist)". So `[2]` would be the `ipaddrlist`, and there's no guaranteed order that the first item in that list is a non-localhost address. In fact, when I run `socket.gethostbyname_ex(socket.gethostname())[2]`, I get `['127.0.1.1', '192.168.1.22']`.

Reply:
Sorry, you are right. Should we use the first IP that is not `127.0.0.1`?

Reply:
Yeah, I guess that's the best approach. Essentially:

* Loop over all the IPs to determine if any are outside the subnet `127.0.0.0/8` (the [ipaddress](https://docs.python.org/3/library/ipaddress.html) module seems to be able to check subnet membership).
* If an IP is outside the subnet, use it.
* If all IPs are in the subnet, use any of them.